data-sources/karma/src/generate-datasets.ts

```diff
@@ -0,0 +1,87 @@
+import { Delegate, Stat } from "./KarmaTypes"
+const Passport = require('../abis/Passport.json')
+const Web3 = require('web3')
+
+/* eslint @typescript-eslint/no-var-requires: "off" */
+const createObjectCsvWriter = require('csv-writer').createObjectCsvWriter
+
+const web3 = new Web3('https://rpc.ankr.com/eth')
+console.info('web3.version:', web3.version)
+
+const PassportContract = new web3.eth.Contract(Passport.abi, '0x3337dac9f251d4e403d6030e18e3cfb6a2cb1333')
+console.info('PassportContract._address:', PassportContract._address)
+
+loadKarmaData()
+
+async function loadKarmaData() {
+  console.info('loadKarmaData')
+
+  const nextId: number = await getNextId()
+  console.info('nextId:', nextId)
+
+  const weekEndDate: Date = new Date()
+  console.info('weekEndDate:', weekEndDate)
+
+  let passportId: number
+  for (passportId = 0; passportId < nextId; passportId++) {
+    console.info('passportId:', passportId)
+
+    const signerAddress: string = await getSigner(passportId)
+    console.info('signerAddress:', signerAddress)
+
+    const writer = createObjectCsvWriter({
+      path: `output/karma-${signerAddress}.csv`,
```

Comment:
What is the logic behind having a file per citizen?

Reply:
@johnmark13 If we include the ETH address in the filename, it's easier to access the data.  When you press a Citizen profile in the Citizen Directory to view more details, we can easily fetch that citizen's data from their file.

Also file versioning becomes less messy this way.